crates/bevy_ecs/src/system/system.rs

```diff
@@ -216,6 +216,220 @@ pub unsafe trait ReadOnlySystem: System {
 /// A convenience type alias for a boxed [`System`] trait object.
 pub type BoxedSystem<In = (), Out = ()> = Box<dyn System<In = In, Out = Out>>;
 
+/// A convenience type alias for a boxed [`ReadOnlySystem`] trait object.
+pub type BoxedReadOnlySystem<In = (), Out = ()> = Box<dyn ReadOnlySystem<In = In, Out = Out>>;
+
+impl<In, Out> System for Box<dyn System<In = In, Out = Out>>
+where
+    In: SystemInput + 'static,
+    Out: 'static,
+{
+    type In = In;
+    type Out = Out;
```

Comment:
Looking at impls for `Box` in `std`, it seems more common to make them generic in the contents of the box, like

```suggestion
impl<S: System + ?Sized> System for Box<S> {
    type In = S::In;
    type Out = S::Out;
````

Then, since `dyn System: System`, that impl covers `Box<dyn System>`.